staging/src/k8s.io/apiserver/pkg/server/genericapiserver.go

```diff
@@ -477,12 +477,13 @@ func (s preparedGenericAPIServer) Run(stopCh <-chan struct{}) error {
 	}
 
 	// pre-shutdown hooks need to finish before we stop the http server
-	preShutdownHooksHasStoppedCh, stopHttpServerCh := make(chan struct{}), make(chan struct{})
+	preShutdownHooksHasStoppedCh := s.lifecycleSignals.PreShutdownHooksCompleted
+	stopHttpServerCh := make(chan struct{})
```

Comment:
I've realized that we already have a `HTTPServerStoppedListening` signal, is not the same?

Reply:
ah, nvm, this is what signals the httpserver to stop, the other is what informs it was stopped